src/mlpack/methods/ann/ffn_impl.hpp

```diff
@@ -78,6 +78,15 @@ double FFN<OutputLayerType, InitializationRuleType, CustomLayers...>::Train(
 {
   ResetData(std::move(predictors), std::move(responses));
 
+  if (optimizer.MaxIterations() != predictors.n_cols)
```

Comment:
I think you should check for
```
optimiser. MaxIterations() >= predictors. n_cols
```

Reply:
Thanks, for pointing this out I was going to make the change before creating a PR but I totally forgot about it. But according to me, the correct relation should be `optimiser. MaxIterations() < predictors. n_cols` because if we're to see #2226 in the comment @shrit mentioned that the issue arises because of the `MaxIterations()` was less than the columns present in the dataset. Correct me if I am wrong.

Reply:
Yes you are right it was a typo from my side